drivers/espi/espi_mchp_xec_v2.c

```diff
@@ -1561,3 +1561,282 @@ static int espi_xec_init(const struct device *dev)
 
 	return ret;
 }
+
+#ifdef CONFIG_FLASH_HAS_DRIVER_ENABLED
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT microchip_xec_espi_maf_flash
+
+#include <zephyr/drivers/flash.h>
+
+#define MAX_FLASH_REQUEST 64U
+
+static const struct device *const espi_dev = DEVICE_DT_GET(DT_NODELABEL(espi0));
+
+struct flash_espi_data {
+	struct k_sem mutex;
+};
+
+static struct flash_espi_data flash_data;
+
+#ifdef CONFIG_FLASH_PAGE_LAYOUT
+static const struct flash_pages_layout espi_flash_layout[] = {
+	{
+	.pages_size = 4096,
+	.pages_count = 1024*512/4096-4096,
+	}
+};
+#endif
+
+bool flash_espi_valid_range(off_t offset, uint32_t len, bool write)
+{
+	return true;
```

Comment:
Need to add validation e.g. SPI flash size, eSPI flash operations will throw an error whenever offset is out-of-range

Reply:
Also need to check whenever range goes across 4K boundaries

Reply:
ack'd.  i was just being lazy on the first cut since i knew i'd be giving valid addresses to the driver.  :)